app/models/storage.rb

```diff
@@ -77,6 +78,10 @@ def ext_management_systems
       :host_storages => {:storage_id => id}).distinct.to_a
   end
 
+  def storage_cluster
+    parent.kind_of?(StorageCluster)? parent.name: nil
```

Comment:
You can have multiple parents here, so you need to limit the scope based on `Manager`